src/build_function.jl

```diff
@@ -164,33 +168,23 @@ Special Keyword Argumnets:
   the `toexpr` function.
 - `checkbounds`: For whether to enable bounds checking inside of the generated
   function. Defaults to false, meaning that `@inbounds` is applied.
-- `linenumbers`: Determines whether the generated function expression retains
-  the line numbers. Defaults to true.
-- `convert_oop`: Determines whether the OOP version should try to convert
-  the output to match the type of the first input. This is useful for
-  cases like LabelledArrays or other array types that carry extra
-  information. Defaults to true.
-- `force_SA`: Forces the output of the OOP version to be a StaticArray.
-  Defaults to `false`, and outputs a static array when the first argument
-  is a static array.
 - `skipzeros`: Whether to skip filling zeros in the in-place version if the
   filling function is 0.
-- `fillzeros`: Whether to perform `fill(out,0)` before the calculations to ensure
-  safety with `skipzeros`.
 """
 function _build_function(target::JuliaTarget, rhss::AbstractArray, args...;
                        expression = Val{true},
                        expression_module = @__MODULE__(),
                        checkbounds = false,
                        postprocess_fbody=ex -> ex,
-                       linenumbers = false,
                        outputidxs=nothing,
                        skipzeros = false,
                        wrap_code = (nothing, nothing),
-                       fillzeros = skipzeros && !(rhss isa SparseMatrixCSC),
```

Comment:
why is `fillzeros` deleted?

Reply:
It isn't used.

Reply:
@shashi what happened to it?

Reply:
It's one of those things that didn't have tests and got removed in the transition... Do we  need it?

Reply:
Yes. I don't think there is a way to test it easily. The functions are all the same from the outside. See there's:

```julia
function f!(du,u)
  du[1] = 0.0
  du[2] = 0.0
  du[3] = 0.0
  du[4] = u[1]
end
```

```julia
function f!(du,u)
  du[4] = u[1]
end
```

```julia
function f!(du,u)
  fill!(du,false)
  du[4] = u[1]
end
```

The other two forms do not do the same thing as the last one: it's a much smaller representation if there are a ton of zeros, but it's also safe in that it does not assume prior zeros. This one cannot just be deleted without consequences. This is probably why the BCR stuff has trouble compiling. Maybe the test to do is on the size of the generated function in a scenario like this, ignoring the actual content of the generated code since that will change over time but just using the size? But please don't just silently delete features without mentioning or documenting it.